homeassistant/components/sensor/homematicip.py

```diff
@@ -0,0 +1,418 @@
+ATTR_APEXCHANGE_CLIENTID = 'apexchange_clientid'
+ATTR_APEXCHANGE_STATE = 'apexchange_state'
+ATTR_ACCESSPOINT = 'accesspoint'
+
+HMIP_UPTODATE = 'up_to_date'
+HMIP_VALVE_DONE = 'adaption_done'
+HMIP_SABOTAGE = 'sabotage'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the HomematicIP sensors devices."""
+    # pylint: disable=import-error, no-name-in-module
+    from homematicip.device import (HeatingThermostat,
+                                    TemperatureHumiditySensorWithoutDisplay,
+                                    TemperatureHumiditySensorDisplay)
+
+    _LOGGER.info('Setting up HomeMaticIP accespoint & generic devices')
+    homeid = discovery_info['homeid']
+    home = hass.data[DOMAIN][homeid]
+    add_devices([HomematicipAccesspoint(hass, home)])
+    if home.devices is None:
+        return False
+    for device in home.devices:
+        add_devices([HomematicipDeviceStatus(hass, home, device)])
+        if isinstance(device, HeatingThermostat):
+            add_devices([HomematicipHeatingThermostat(hass, home, device)])
+        if isinstance(device, TemperatureHumiditySensorWithoutDisplay):
+            add_devices([HomematicipSensorThermometer(hass, home, device)])
+            add_devices([HomematicipSensorHumidity(hass, home, device)])
+        if isinstance(device, TemperatureHumiditySensorDisplay):
+            add_devices([HomematicipSensorThermometer(hass, home, device)])
+            add_devices([HomematicipSensorHumidity(hass, home, device)])
+    return True
+
+
+class HomematicipGenericDevice(Entity):
+    """Representation of an HomematicIP generic device."""
+
+    def __init__(self, hass, home, device, signal=None):
+        """Initialize the generic device."""
+        self.hass = hass
+        self._home = home
+        self._device = device
+
+        @callback
+        def event_device(event):
+            """Handle device state changes."""
+            _LOGGER.debug('Event Device: %s', self._device.label)
+            self.schedule_update_ha_state(True)
+
+        hass.bus.listen(EVENT_DEVICE_CHANGED, event_device)
+
+    def _name(self, addon=''):
+        """Return the name of the device."""
+        name = ''
+        if self._home.label != '':
+            name += self._home.label + ' '
+        name += self._device.label
+        if addon != '':
+            name += ' ' + addon
+        return name
+
+    @property
+    def name(self):
+        """Return the name of the generic device."""
+        return self._name()
+
+    @property
+    def icon(self):
+        """Return the icon of the generic device."""
+        return 'mdi:hexagon-outline'
+
+    @property
+    def state(self):
+        """Return the state of the generic device."""
+        return 'empty'
+
+    @property
+    def force_update(self):
+        """Force update."""
+        return True
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return True
```

Comment:
We should not poll, you have code to update the entity when events happen from homematic IP